server/src/main/java/org/elasticsearch/index/mapper/KeywordFieldMapper.java

```diff
@@ -305,6 +310,31 @@ protected BytesRef indexedValueForSearch(Object value) {
             if (value instanceof BytesRef) {
                 value = ((BytesRef) value).utf8ToString();
             }
+            if (value instanceof String) {
+                // best effort to not pass the wildcard characters and escaped characters through #normalize
```

Comment:
We should do that only for wildcard queries. For other term-based queries these characters should remain in the normalization so I'd move this code to `StringFieldType#wildcardQuery` ?